packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -0,0 +1,182 @@
+    justifyContent: 'space-between',
+    paddingVertical: spacing(3),
+    paddingHorizontal: spacing(6),
+    borderBottomColor: palette.neutralLight8,
+    borderBottomWidth: 1
+  },
+  lastRow: {
+    borderBottomWidth: 0
+  },
+  greyRow: {
+    backgroundColor: palette.neutralLight10
+  },
+  summaryTitle: {
+    letterSpacing: 1
+  },
+  payToUnlockTitleContainer: {
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    gap: spacing(2),
+    marginBottom: spacing(2)
+  },
+  buy: {
+    fontSize: typography.fontSize.large,
+    lineHeight: typography.fontSize.large * 1.5
+  },
+  button: {
+    padding: spacing(4),
+    height: spacing(12)
+  }
+}))
+
+export const PremiumTrackPurchaseDrawer = () => {
+  const styles = useStyles()
+  const { neutralLight2, specialLightGreen1 } = useThemeColors()
+  const { data } = useDrawer('PremiumTrackPurchase')
+  const { trackId } = data
+  const track = useSelector((state) => getTrack(state, { id: trackId }))
+  const user = useSelector((state) => getUser(state, { id: track?.owner_id }))
+
+  const handleConfirmPress = useCallback(() => {
+    console.log('buy button pressed')
+  }, [])
+
+  const { premium_conditions: premiumConditions } = track ?? {}
+
+  if (!track || !isPremiumContentUSDCPurchaseGated(premiumConditions) || !user)
+    return null
+
+  const price = formatUSDCWeiToUSDString(premiumConditions.usdc_purchase.price)
+
+  return (
+    <NativeDrawer drawerName={PREMIUM_TRACK_PURCHASE_MODAL_NAME}>
+      <View style={styles.drawer}>
+        <View style={styles.titleContainer}>
+          <IconCart fill={neutralLight2} />
+          <Text style={styles.title}>{messages.title}</Text>
+        </View>
+        <TrackDetailsTile trackId={track.track_id} />
+        <View style={styles.summaryContainer}>
+          <View style={[styles.summaryRow, styles.greyRow]}>
+            <Text
+              weight='bold'
+              textTransform='uppercase'
+              style={styles.summaryTitle}
+            >
+              {messages.summary}
+            </Text>
+          </View>
+          <View style={styles.summaryRow}>
+            <Text>{messages.artistCut}</Text>
+            <Text>{messages.price(price)}</Text>
+          </View>
+          <View style={styles.summaryRow}>
+            <Text>{messages.audiusCut}</Text>
+            <Text>{messages.always0}</Text>
+          </View>
+          <View style={[styles.summaryRow, styles.lastRow, styles.greyRow]}>
+            <Text weight='bold'>{messages.youPay}</Text>
+            <Text weight='bold' color='secondary'>
+              {messages.price(price)}
+            </Text>
+          </View>
+        </View>
+        <View>
+          <View style={styles.payToUnlockTitleContainer}>
+            <Text weight='heavy' textTransform='uppercase' fontSize='small'>
+              {messages.payToUnlock}
+            </Text>
+            <LockedStatusBadge locked={true} />
```

Comment:
I started a comment here to recommend omitting the value on boolean JSX attributes when they are true. I know there's an ESLint rule for it (though I don't think we use it). But I didn't know _why_ that practice is recommended.
Turns out it's because of [how booleans work in native HTML components](https://getfrontend.tips/omit-values-of-html-boolean-attributes/)

TL/DR: providing a value can often not result in the behavior you expect. Though we don't quite have the same problem here, I think the idea is to get in the habit of just doing `<... locked />` so that when you _do_ use a native component, the consistent pattern prevents errors.

Anyway, not in our style guide, so feel free to ignore 😅

Reply:
I support this in the styleguide :)